tests/python_package_test/test_engine.py

```diff
@@ -4611,3 +4612,22 @@ def test_bagging_by_query_in_lambdarank():
     ndcg_score_no_bagging_by_query = gbm_no_bagging_by_query.best_score["valid_0"]["ndcg@5"]
     assert ndcg_score_bagging_by_query >= ndcg_score - 0.1
     assert ndcg_score_no_bagging_by_query >= ndcg_score - 0.1
+
+
+def test_equal_datasets_from_row_major_and_col_major_data(tmp_path):
```

Comment:
As this test now doesn't include `train()`, I think it should go to `test_basic.py` file.

Reply:
Sorry. Moved in 38c6786